### PR TITLE
[daint, dom] Integration of ADIOS with Cray wrappers and refinement

### PR DESCRIPTION
**Pull request by omlins**

### CODE REVIEW

**Comment by omlins** [comment]
It successfully builds using the wrappers - all the compile an linking flags are added automatically:
omlins@dom101:~/10_io_perf/0_adios2_helloworld> cc helloBPWriter.c
omlins@dom101:~/10_io_perf/0_adios2_helloworld> CC helloBPWriter.cpp
omlins@dom101:~/10_io_perf/0_adios2_helloworld> ftn helloBPWriter.F90
omlins@dom101:~/10_io_perf/0_adios2_helloworld> CC helloBPWriter.c -fpermissive
helloBPWriter.c: In function 'int main(int, char**)':
helloBPWriter.c:54:22: warning: invalid conversion from 'void*' to 'float*' [-fpermissive]
     myFloats = malloc(sizeof(float) * Nx);
                ~~~~~~^~~~~~~~~~~~~~~~~~~~
omlins@dom101:~/10_io_perf/0_adios2_helloworld>

**Review by jfavre — CHANGES_REQUESTED** [review]
should moduleclass of blosc be changed from 'vis' to 'tools' to match similar other modules such as the bzip2?

Other than that, it lgtm

**Comment by omlins** [comment]
> should moduleclass of blosc be changed from 'vis' to 'tools' to match similar other modules such as the bzip2?
> 
> Other than that, it lgtm

@gppezzi, what do you think?

Besides that, note that this is still WIP as I am adding more compression codecs.

**Comment by omlins** [comment]
What is missing is to set the `moduleclass` - maybe `lib` could also be used... Besides that, everything should be fine.

**Comment by omlins** [comment]
@jfavre and @gppezzi : with the exception of the moduleclass issue it now it is completely ready for merging from my point of view. Please share what you think about this issue (when to use tools, vis and lib) in order to merge it afterwards.

File: easybuild/easyconfigs/a/adios/adios-2.6.0-CrayGnu-20.08.eb [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# contributed by Jean Favre (CSCS)
+# contributed by Jean Favre (CSCS); extended by Samuel Omlin (CSCS)
```
Comment (gppezzi) [inline easybuild/easyconfigs/a/adios/adios-2.6.0-CrayGnu-20.08.eb:1]: I've just noticed that the file name does not match the toolchain (case): CrayGNU

**Review by gppezzi — APPROVED** [review]

File: easybuild/easyconfigs/a/adios/adios-2.6.0-CrayGnu-20.08.eb (hunk 1 above)
Comment (jfavre) [reply easybuild/easyconfigs/a/adios/adios-2.6.0-CrayGnu-20.08.eb:1]: @omlins Can you fix the filename? Thanks

Comment (omlins) [reply easybuild/easyconfigs/a/adios/adios-2.6.0-CrayGnu-20.08.eb]: Done.

**Comment by omlins** [comment]
Concerning the moduleclasses: I think the primary purpose of all the software dealt with in this PR with is to be used as a library. Thus, I changed the module class for all to `lib`. I hope you agree with this. Besides that @gppezzi pointed out that this is essentially cosmetics as long as we don't use lmod and does not have any practical consequences for us.

**Comment by omlins** [comment]
@jfavre please let us know if you agree with this.

**Review by jfavre — APPROVED** [review]

**Pull request merged**